_posts/2022-09-09-hhvm-4.168.markdown

```diff
@@ -0,0 +1,50 @@
+---
+title: "HHVM 4.168"
+layout: post
+author: alexeyt
+category: blog
+---
+
+HHVM 4.168 is released! This release has long term support, so:
+- this release will be supported for approximately 48 weeks
+- the next LTS will be released in approximately 24 weeks
+- HHVM 4.153 (LTS) and HHVM 4.165&ndash;167 remain supported
+- support has ended for 4.128 (LTS) and 4.164
+
+# Status of ubuntu 22.04 support
+
+We had intended for this release to have ubuntu 22.04 support, but we are 
+releasing without that support for now. Due to library version compatibility
+issues, we need more time to get this support working, and we did not want 
+to delay the release for other platforms. We intend to make a 4.168.1 
+release with 22.04 support as soon as we get builds working on 22.04.
+
+# Highlights
+
+- Fix a typechecker bug that could infer some shape keys as optional when 
+  they were in fact always present.
+  ([commit](https://github.com/facebook/hhvm/commit/3d86eed063))
```

Comment:
I believe this is for the dict-to-shape static analysis tool, and doesn't affect the normal typechecker AIUI.